test/server/cards/03_TWI/leaders/CaptainRexFightingForHisBrothers.spec.ts

```diff
@@ -0,0 +1,81 @@
+describe('Captain Rex, Fighting for his Brothers', function () {
+    integration(function (contextRef) {
+        beforeEach(function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    leader: 'captain-rex#fighting-for-his-brothers',
+                    groundArena: ['advanced-recon-commando'],
+                    resources: 5
+                },
+                player2: {
+                    groundArena: ['droid-commando']
+                }
+            });
+        });
+
+        describe('Captain Rex\'s leader undeployed ability', function () {
+            it('should create a Clone Tropper when a friendly unit attacked this phase', function () {
+                const { context } = contextRef;
+
+                // Leader ability should be available, but should not create a Clone Trooper since no friendly unit has attacked this phase
+                context.player1.clickCard(context.captainRex);
+                context.player1.clickPrompt('If a friendly unit attacked this phase, create a Clone Trooper token.');
+
+                expect(context.captainRex.exhausted).toBeTrue();
+                expect(context.player1.exhaustedResourceCount).toBe(2);
+                expect(context.player1.findCardsByName('clone-trooper').length).toBe(0);
+
+                context.moveToNextActionPhase();
+
+                // Leader ability should be available, but should not create a Clone Trooper since an enemy unit attacked this phase
```

Comment:
You could remove the above test where nothing attacks at all and just do this one since it covers both cases

Reply:
sounds good, I am still feeling out my way on how much and what we want to test with limited GitHub Actions minutes.